src/Kuzzle.ts

```diff
@@ -526,15 +514,18 @@ export class Kuzzle extends KuzzleEventEmitter {
    * @param request
    * @param options - Optional arguments
    */
-  query (request: RequestPayload = {}, options: JSONObject = {}): Promise<ResponsePayload> {
-    if (typeof request !== 'object' || Array.isArray(request)) {
-      throw new Error(`Kuzzle.query: Invalid request: ${JSON.stringify(request)}`);
+  query (req: RequestPayload = {}, opts: JSONObject = {}): Promise<ResponsePayload> {
+    if (typeof req !== 'object' || Array.isArray(req)) {
+      throw new Error(`Kuzzle.query: Invalid request: ${JSON.stringify(req)}`);
     }
 
-    if (typeof options !== 'object' || Array.isArray(options)) {
-      throw new Error(`Kuzzle.query: Invalid "options" argument: ${JSON.stringify(options)}`);
+    if (typeof opts !== 'object' || Array.isArray(opts)) {
+      throw new Error(`Kuzzle.query: Invalid "options" argument: ${JSON.stringify(opts)}`);
     }
 
+    const request = { ...req };
+    const options = { ...opts };
```

Comment:
(nitpicking) or you could make a deep-copy, as it is safer in case users modify the referenced objects before they are submitted again, if they get queued

```suggestion
    const request = JSON.parse(JSON.stringify(req));
    const options = JSON.parse(JSON.stringify(opts));
```

Reply:
That's what I was doing at first but then I thought about the performance drop and I sticked to the light copy. 
The query method is only mutating first level properties so IMHO it's enough and if users mutate there own payloads I think it's not our responsibility anymore

Reply:
About performances, I believe the JSON.stringify/JSON.parse method performs at roughly a couple millions instructions per second, so we should be quite safe there.